devmand/gateway/src/devmand/channels/cli/IoConfigurationBuilder.cpp

```diff
@@ -76,31 +76,50 @@ chrono::seconds IoConfigurationBuilder::loadTimeout(
   }
 }
 
-shared_ptr<Cli> IoConfigurationBuilder::createAll(
-    function<shared_ptr<Cli>(
-        shared_ptr<IOThreadPoolExecutor>,
-        shared_ptr<ConnectionParameters>)> underlyingCliLayerFactory,
+shared_ptr<Cli> IoConfigurationBuilder::createAllUsingFactory(
     shared_ptr<CliCache> commandCache) {
   shared_ptr<folly::ThreadWheelTimekeeper> timekeeper =
       make_shared<folly::ThreadWheelTimekeeper>(); // TODO use singleton when
                                                    // folly is initialized
 
   // TODO make Executor sharing configurable
-  function<shared_ptr<Cli>()> cliFactory = [underlyingCliLayerFactory,
-                                            params = connectionParameters,
-                                            timekeeper,
-                                            commandCache]() -> shared_ptr<Cli> {
-    shared_ptr<IOThreadPoolExecutor> executor =
-        make_shared<IOThreadPoolExecutor>(
-            10, std::make_shared<NamedThreadFactory>("persession"));
-    return getIo(
-        params,
-        underlyingCliLayerFactory(executor, params),
-        timekeeper,
-        commandCache);
-  };
-
-  // create reconnecting cli
+  shared_ptr<IOThreadPoolExecutor> executor = make_shared<IOThreadPoolExecutor>(
+      10, std::make_shared<NamedThreadFactory>("persession"));
+
+  function<SemiFuture<shared_ptr<Cli>>()> cliFactory =
+      [executor, params = connectionParameters, timekeeper, commandCache]() {
+        return createPromptAwareCli(executor, params)
+            .via(executor.get())
+            .thenValue(
+                [executor, params, commandCache, timekeeper](
+                    shared_ptr<Cli> sshCli) -> shared_ptr<Cli> {
+                  MLOG(MDEBUG) << "[" << params->id
+                               << "] "
+                                  "Creating cli layers rcclli, ttcli, qcli";
+                  // create caching cli
+                  const shared_ptr<ReadCachingCli>& rccli =
+                      std::make_shared<ReadCachingCli>(
+                          params->id, sshCli, commandCache);
+                  // create timeout tracker
+                  shared_ptr<TimeoutTrackingCli> ttcli =
+                      TimeoutTrackingCli::make(
+                          params->id,
+                          rccli,
+                          timekeeper,
+                          std::make_shared<folly::CPUThreadPoolExecutor>(
+                              5, std::make_shared<NamedThreadFactory>("ttcli")),
```

Comment:
does ttcli need 5 threads?

Reply:
This commit is not changing threadpools. I believe you are working on that.